src/Backend/Modules/Pages/Domain/Page/PageRepository.php

```diff
@@ -222,6 +247,52 @@ public function getFirstChild(int $pageId, string $status, string $language): ?P
             ->getOneOrNullResult();
     }
 
+    public function getRevisionIdsToKeep(int $id, int $rowsToKeep): array
+    {
+        $result = $this
+            ->createQueryBuilder('p')
+            ->select('p.revisionId')
+            ->where('p.id = :id')
+            ->andWhere('p.status = :status')
+            ->orderBy('p.editedOn', 'DESC')
+            ->setMaxResults($rowsToKeep)
+            ->setParameters(
+                [
+                    'id' => $id,
+                    'status' => Page::ARCHIVE,
+                ]
+            )
+            ->getQuery()
+            ->getResult(AbstractQuery::HYDRATE_SCALAR);
+
+        return array_column($result, 'revisionId');
+    }
+
+    public function getRevisionIdsToDelete(int $id, string $status, array $revisionsToKeep): array
```

Comment:
I would use a value object for the status

Reply:
Changed in f61a8e90